cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -684,34 +692,43 @@ private void restartClouderaManagementServices(MgmtServiceResourceApi mgmtServic
+                stack, v31Client, restartCommand.getId());
         handlePollingResult(hostTemplatePollingResult.getPollingResult(), "Cluster was terminated while waiting for service restart",
                 "Timeout while Cloudera Manager was restarting services.");
     }
 
     @VisibleForTesting
     void pollDeployConfig(BigDecimal commandId) throws CloudbreakException {
         ExtendedPollingResult hostTemplatePollingResult = clouderaManagerPollingServiceProvider.startPollingCmClientConfigDeployment(
-                stack, apiClient, commandId);
+                stack, v31Client, commandId);
         handlePollingResult(hostTemplatePollingResult, "Cluster was terminated while waiting for config deploy",
                 "Timeout while Cloudera Manager was config deploying services.");
     }
 
     @VisibleForTesting
     void pollRefresh(ApiCommand restartCommand) throws CloudbreakException {
         ExtendedPollingResult hostTemplatePollingResult = clouderaManagerPollingServiceProvider.startPollingCmConfigurationRefresh(
-                stack, apiClient, restartCommand.getId());
+                stack, v31Client, restartCommand.getId());
         handlePollingResult(hostTemplatePollingResult, "Cluster was terminated while waiting for service refresh",
                 "Timeout while Cloudera Manager was refreshing services.");
     }
 
     private void applyHostGroupRolesOnUpscaledHosts(ApiHostRefList body, String hostGroupName) throws ApiException, CloudbreakException {
         if (body.getItems() != null && !body.getItems().isEmpty()) {
             LOGGER.debug("Applying host template on upscaled hosts. Host group: [{}]", hostGroupName);
-            HostTemplatesResourceApi templatesResourceApi = clouderaManagerApiFactory.getHostTemplatesResourceApi(apiClient);
-            ApiCommand applyHostTemplateCommand = templatesResourceApi.applyHostTemplate(stack.getName(), hostGroupName, START_ROLES_ON_UPSCALED_NODES, body);
+            ClouderaManagerRepo clouderaManagerRepoDetails = clusterComponentProvider.getClouderaManagerRepoDetails(stack.getCluster().getId());
+            ApiCommand applyHostTemplateCommand;
+            if (isVersionNewerOrEqualThanLimited(clouderaManagerRepoDetails::getVersion, CLOUDERAMANAGER_VERSION_7_10_0)) {
+                HostTemplatesResourceApi hostTemplatesResourceApi = clouderaManagerApiFactory.getHostTemplatesResourceApi(v52Client);
+                applyHostTemplateCommand = hostTemplatesResourceApi
+                        .applyHostTemplate(stack.getName(), hostGroupName, true, START_ROLES_ON_UPSCALED_NODES, body);
+            } else {
+                HostTemplatesResourceApi hostTemplatesResourceApi = clouderaManagerApiFactory.getHostTemplatesResourceApi(v31Client);
+                applyHostTemplateCommand = hostTemplatesResourceApi
+                        .applyHostTemplate(stack.getName(), hostGroupName, false, START_ROLES_ON_UPSCALED_NODES, body);
+            }
```

Comment:
TBH I still do not see how this additional check will help here. The invoked method is the same in both cases, the only difference will be in the CM API version in the target URL. :smile: